spec/models/article_spec.rb

```diff
@@ -136,6 +135,32 @@ def build_and_validate_article(*args)
       end
     end
 
+    describe "#canonical_url_must_be_unique" do
+      let(:url) { "http://example.com" }
+
+      it "is valid when the canonical url is nil" do
+        another_article = build(:article)
+
+        expect(another_article).to be_valid
+      end
+
+      it "is not valid" do
+        body_markdown = "---\ntitle: Title\ncanonical_url: #{url}\npublished: true\n---\n\n"
+        create(:article, body_markdown: body_markdown)
+        another_article = build(:article, body_markdown: body_markdown)
+
+        expect(another_article).not_to be_valid
```

Comment:
I prefer being somewhat more explicit in my assertions (i.e. checking for a specific error condition), so this test can't accidentally remain green.